### PR TITLE
chore: add new property for recommendations in order completed event

### PR DESCRIPTION
**Pull request by waheedahmed**
## Description
We (Activate Squad) working on a POC for personalized recommendations and needed to add a new property in the existing purchase completed event since this event is a key metric in this POC.

## Supporting information
[VAN-987](https://2u-internal.atlassian.net/browse/VAN-987)

### CODE REVIEW

File: ecommerce/courses/utils.py [hunk 1]
```diff
@@ -152,3 +151,17 @@ def get_certificate_type_display_value(certificate_type):
         raise ValueError('Certificate Type [{}] not found.'.format(certificate_type))
 
     return display_values[certificate_type]
+
+
+def get_personalized_recommendation(request, course_id):
```
Comment (waheedahmed) [inline ecommerce/courses/utils.py]: Rename: `check_is_personalized_recommendation`

File: ecommerce/extensions/checkout/signals.py [hunk 2]
```diff
@@ -33,6 +33,33 @@ def track_completed_order(sender, order=None, **kwargs):  # pylint: disable=unus
     if order.total_excl_tax <= 0:
         return
 
+    request = kwargs.get('request', None)
```
Comment (waheedahmed) [inline ecommerce/extensions/checkout/signals.py]: Could this be a payment processor request object? If yes, we won't find that cookie in it.

File: ecommerce/courses/utils.py [hunk 3]
```diff
@@ -152,3 +151,17 @@ def get_certificate_type_display_value(certificate_type):
...
+    """
+    Returns the personalized recommendation value from the cookie.
+    """
+    recommended_courses = \
```
Comment (julianajlk) [inline ecommerce/courses/utils.py]: The purpose of the escape character here is because the line is too long?

**Review by waheedahmed — COMMENTED** [review]
@shafqatfarhan One change and a question otherwise looks good.

File: ecommerce/extensions/checkout/signals.py (hunk 2 above)
Comment (waheedahmed) [reply ecommerce/extensions/checkout/signals.py]: Also, can we move this into `get_personalized_recommendation` since it's not used anywhere else here?

**Review by julianajlk — COMMENTED** [review]
Not sure how `properties.products` will be used for this use case, will it be an issue when `is_personalized_recommendation` is None and the key doesn't exist/wasn't set because it's None? 

Are there any unit tests?

File: ecommerce/courses/utils.py [hunk 4]
```diff
@@ -152,3 +153,16 @@ def get_certificate_type_display_value(certificate_type):
...
+    """
+    Returns the personalized recommendation value from the cookie.
+    """
+    if request and course:
+        recommendations = request.COOKIES.get('edx-user-personalized-recommendation', None)
+        if recommendations:
+            recommendations = json.loads(unquote(recommendations))
+            if course.id in recommendations['course_keys']:
+                return recommendations['is_personalized_recommendation']
+    return None
```
Comment (waheedahmed) [inline ecommerce/courses/utils.py:168]: Add an empty line above this.

File: ecommerce/courses/utils.py (hunk 1 above)
Comment (shafqatfarhan) [reply ecommerce/courses/utils.py]: done

**Review by waheedahmed — COMMENTED** [review]
One minor nit otherwise approved 👍

File: ecommerce/extensions/checkout/signals.py (hunk 2 above)
Comment (shafqatfarhan) [reply ecommerce/extensions/checkout/signals.py]: I have verified it locally, the request object we get in `track_completed_order` contains the cookies we require.

File: ecommerce/courses/utils.py (hunk 3 above)
Comment (shafqatfarhan) [reply ecommerce/courses/utils.py]: Yes. I have also updated the variable name to make the value assignment in a single line.

**Comment by shafqatfarhan** [comment]
> Not sure how `properties.products` will be used for this use case, will it be an issue when `is_personalized_recommendation` is None and the key doesn't exist/wasn't set because it's None?
> 
> Are there any unit tests?

In our use case, we need to add a new property `is_personalized_recommendation` only if a course ID matches in the cookie. We are not setting this property if `is_personalized_recommendation` is None. This property is required in the Segment event for measuring analytics for our Recommendations POC.

Also, added a unit test to cover the logic added.

Thanks!

**Comment by shafqatfarhan** [comment]
@julianajlk PR is updated. Please review and merge (if approved). Thanks

**Review by shafqatfarhan — APPROVED** [review]
👍

**Review by julianajlk — APPROVED** [review]
LGTM, thanks for updating the tests!

**Pull request merged**